lib/commanded/event/handler.ex

```diff
@@ -1022,21 +1264,28 @@ defmodule Commanded.Event.Handler do
 
         :timer.sleep(delay)
 
-        handle_event(failed_event, context, state)
+        retry_fun.(context, state)
 
       {:retry, delay, context} when is_map(context) and is_integer(delay) and delay >= 0 ->
         # Retry the failed event after waiting for the given delay, in milliseconds
         Logger.info(describe(state) <> " is retrying failed event after #{inspect(delay)}ms")
 
         :timer.sleep(delay)
 
-        handle_event(failed_event, context, state)
+        retry_fun.(context, state)
 
       :skip ->
         # Skip the failed event by confirming receipt
         Logger.info(describe(state) <> " is skipping event")
+        state = confirm_receipt([maybe_failed_event], state)
```

Comment:
We call `confirm_receipt` in a few spots where we have just a single event. How about normalizing to lists in confirm_receipt so callers don't have to do the wrapping?